src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -253,6 +262,21 @@ public void testRequiredTokenIsEmptyIntArray() throws Exception {
         }
     }
 
+    @Test
+    public void testBehaviourWithZeroChecks() throws Exception {
+        final TreeWalker treeWalker = new TreeWalker();
+        final String errMsg = "There should be no parsing or walking ASTs if there are no checks";
+        mockStatic(TreeWalker.class);
+        when(TreeWalker.parse(any(FileContents.class)))
+                .thenThrow(new IllegalStateException(errMsg));
```

Comment:
Strange way of verification. Can't something like [this](https://static.javadoc.io/org.mockito/mockito-core/2.7.21/org/mockito/Mockito.html#4) be used?